File: thesum.c
#include <stdio.h>

int main()
{
	FILE *f;	 
	int num = 0;
	int f_num = 0;
	int somma;

	f = fopen("sum.dat","r");

	if(f) {
		fscanf(f, "%d", &f_num);
		fprintf(stdout, "Lettura dal file sum.dat in corso...\n");
		printf("Valore corrente: %d\n\n", f_num);

		printf("Inserisca il valore da sommare: ");

		fclose(f);
	}

	else{
		fprintf(stderr, "Il file sum.dat non esiste.\nCreare un nuovo valore: ");
	}

	scanf("%d", &num);

	somma = num + f_num;

	f = fopen("sum.dat","w");

	fprintf(f, "%d\n", somma);

	printf("\nHo letto il valore %d\n", somma);

	fprintf(stdout, "Salvataggio nel file sum.dat in corso...\n");

	fclose(f);

	return 0;
}

File: thesum.py
import os

f = 'sum.dat'
f_num = 0

if os.path.isfile(f):
	f = open(f, 'r');
	string = f.readline()
	f_num = int(string)
	print('Lettura dal file sum.dat in corso...')
	print('Valore corrente: ', f_num)
	print('Faro la somma di un nuovo valore.')

	f.close()
else:
	print('Il file sum.dat non esiste.')

num = input('Creare un nuovo valore: ')
num = int(num)
somma = f_num + num
somma = str(somma)

f = open('sum.dat', 'w')

f.write(somma)

print('Ho letto il valore ', somma)

print('Salvataggio nel file sum.dat in corso...')

f.close()
